src/main/scala/org/hammerlab/guacamole/reads/MDTagUtils.scala

```diff
@@ -75,7 +76,42 @@ object MDTagUtils {
       }
     })
 
-    reference.elems
+    reference.toArray
+  }
+
+  def getReference(read: MappedRead, allowNBase: Boolean): Seq[Byte] = {
+    getReference(read.mdTag, read.sequence, read.cigar, read.start, allowNBase)
+  }
+
+  /**
+   * Rebuilds the reference from a set of overlapping and sorted reads
```

Comment:
Some comments:
 * It feels weird that this function doesn't take a start and an end locus to return the reference sequence for. Is the idea that the caller would look at the reads first to know where it starts? That seems potentially confusing e.g. in the case of clipped reads.
 * If you do move to taking a start and an end locus, might as well build up the sequence directly in an array (instead of a buffer), to avoid the copy.
 * If multiple reads are overlapping a locus, it feels like we should assert that they agree on the reference bases.
 * Consider an assertion to force that the reads are sorted